Client/Frontend/Browser/Onboarding/Welcome/WelcomeViewCallout.swift

```diff
@@ -0,0 +1,478 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import UIKit
+import SnapKit
+import BraveUI
+import Shared
+import pop
+
+enum WelcomeViewCalloutState {
+    case welcome(title: String)
+    case privacy(title: String, details: String, buttonTitle: String, action: () -> Void)
+    case defaultBrowser(title: String, details: String, primaryButtonTitle: String, secondaryButtonTitle: String, primaryAction: () -> Void, secondaryAction: () -> Void)
+    case defaultBrowserWarning(title: String, details: String, primaryButtonTitle: String, secondaryButtonTitle: String, primaryAction: () -> Void, secondaryAction: () -> Void)
+    case ready(title: String, details: String, moreDetails: String)
+}
+
+class WelcomeViewCallout: UIView {
+    private struct DesignUX {
+        static let padding = 20.0
+        static let contentPadding = 30.0
+        static let cornerRadius = 16.0
+    }
+    
+    private let backgroundView = RoundedBackgroundView(cornerRadius: DesignUX.cornerRadius)
+    
+    private let arrowView = CalloutArrowView().then {
+        $0.backgroundColor = .secondaryBraveBackground
+    }
+    
+    private let contentView = UIStackView().then {
+        $0.axis = .vertical
+        $0.layoutMargins = UIEdgeInsets(equalInset: DesignUX.contentPadding)
+        $0.isLayoutMarginsRelativeArrangement = true
+    }
+    
+    // MARK: - Content
+    private let titleLabel = UILabel().then {
+        $0.textColor = .bravePrimary
+        $0.textAlignment = .center
+        $0.numberOfLines = 0
+        $0.setContentHuggingPriority(.required, for: .vertical)
+        $0.setContentCompressionResistancePriority(.required, for: .vertical)
+    }
+    
+    private let detailsLabel = UILabel().then {
+        $0.textColor = .bravePrimary
+        $0.textAlignment = .left
+        $0.numberOfLines = 0
+        $0.setContentHuggingPriority(.required, for: .vertical)
+        $0.setContentCompressionResistancePriority(.required, for: .vertical)
+    }
+    
+    private let primaryButton = RoundInterfaceButton(type: .custom).then {
+        $0.setTitleColor(.white, for: .normal)
+        $0.backgroundColor = .braveBlurple
+        $0.setTitle(" ", for: .normal)
```

Comment:
Maybe add comment above why one space string is needed here

Reply:
Unnecessary addition removed 
Fixed at https://github.com/brave/brave-ios/pull/4508/commits/9f72c586778075cc4dc78b26127c110ec6ce4bb8